File: integration-tests/iac/lambdas/node/rds_mysql.mjs
import mysql from 'mysql2/promise';

export async function handler(event, context) {
    const connection = await mysql.createConnection({
        host: process.env.DB_HOST,
        port: parseInt(process.env.DB_PORT || '3306'),
        database: process.env.DB_NAME,
        user: process.env.DB_USER,
        password: process.env.DB_PASSWORD,
        ssl: { rejectUnauthorized: false },
    });

    try {
        await connection.execute(`
            CREATE TABLE IF NOT EXISTS test_entries (
                id INT AUTO_INCREMENT PRIMARY KEY,
                request_id TEXT NOT NULL,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
            )
        `);

        const [insertResult] = await connection.execute(
            'INSERT INTO test_entries (request_id) VALUES (?)',
            [context.awsRequestId],
        );

        const [rows] = await connection.execute(
            'SELECT * FROM test_entries WHERE request_id = ?',
            [context.awsRequestId],
        );

        console.log(`Inserted and read back entry for request: ${context.awsRequestId}`);

        return {
            statusCode: 200,
            body: JSON.stringify({
                inserted: { id: insertResult.insertId, request_id: context.awsRequestId },
                read: rows[0],
            }),
        };
    } finally {
        await connection.end();
    }
}

File: integration-tests/iac/lambdas/python-db/rds_mysql.py
import json
import os
import pymysql


def handler(event, context):
    conn = pymysql.connect(
        host=os.environ["DB_HOST"],
        port=int(os.environ.get("DB_PORT", "3306")),
        database=os.environ["DB_NAME"],
        user=os.environ["DB_USER"],
        password=os.environ["DB_PASSWORD"],
        ssl={"ssl": True},
    )

    try:
        with conn.cursor() as cur:
            cur.execute(
                """
                CREATE TABLE IF NOT EXISTS test_entries (
                    id INT AUTO_INCREMENT PRIMARY KEY,
                    request_id TEXT NOT NULL,
                    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
                )
                """
            )
            conn.commit()

            cur.execute(
                "INSERT INTO test_entries (request_id) VALUES (%s)",
                (context.aws_request_id,),
            )
            conn.commit()

            cur.execute(
                "SELECT * FROM test_entries WHERE request_id = %s",
                (context.aws_request_id,),
            )
            read = cur.fetchone()

        print(f"Inserted and read back entry for request: {context.aws_request_id}")

        return {
            "statusCode": 200,
            "body": json.dumps(
                {
                    "inserted": {"request_id": context.aws_request_id},
                    "read": str(read),
                }
            ),
        }
    finally:
        conn.close()
